### PR TITLE
Make sure trusted_proxies is an array

### LINKED ISSUE #26256: proxy header warning disappear when trusted_proxies is set to an invalid value
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. setup reverse proxy with correct x-forwarded-for headers
2. add trusted proxy in config.php: `'trusted_proxies' => '<ip-of-proxy>',` to get the warning removed:
grafik

### Expected behaviour
Nextcloud should log the client's ip on events like failed to login for example because there is no warning anymore so it seems to work.

### Actual behaviour
Nextcloud will log the ip of the proxy instead evaluating the x-forwarded-for header. This is because we did not set trusted_proxies as an array:
```
  'trusted_proxies' => 
  array (
    0 => '<ip-of-proxy>',
  ),
```
### Server configuration

**Operating system:**
ubuntu

**Web server:**
nginx

**PHP version:**
7.4

**Nextcloud version:** (see Nextcloud admin page)
20.0.7

**Updated from an older Nextcloud/ownCloud or fresh install:**
updated

**Where did you install Nextcloud from:**
nextcloud.com

**Signing status:**
<details>
<summary>Signing status</summary>
No errors have been found.
```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
<?php
$CONFIG = array (
  'passwordsalt' => 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
  'secret' => 'xxxxxxxxxxxxxxxxxxxxxxxxxx',
  'trusted_domains' => 
  array (
    0 => 'xxxxxxxxxxxxxxxx',
  ),
  'defaultapp' => 'apporder',
  'datadirectory' => '/nextcloud-data',
  'overwrite.cli.url' => 'xxxxxxxxxxxxxxxxxxxxx',
  'overwriteprotocol' => 'https',
  'dbtype' => 'mysql',
  'version' => '20.0.7.1',
  'dbname' => 'xxxxxxxxxxxxxxx',
  'dbhost' => '1.2.3.4',
  'dbtableprefix' => 'nc_',
  'dbuser' => 'xxxxxxxxxxxxxxxx',
  'dbpassword' => 'xxxxxxxxxxxxxxx',
  'logtimezone' => 'Europe/Berlin',
  'logfile' => '/var/log/nextcloud.log',
  'loglevel' => 1,
  'log_authfailip' => true,
  'installed' => true,
  'maintenance' => false,
  'singleuser' => false,
  'mail_smtpmode' => 'smtp',
  'mail_smtpauth' => 1,
  'mail_smtphost' => 'xxxxxxxxxxxxxxx.de',
  'mail_smtpport' => '465',
  'mail_from_address' => 'xxxxxxxxxxxxxxxxxx',
  'mail_domain' => 'xxxxxxxxxxxxxx',
  'mail_smtpname' => 'xxxxxxxxxxxxxxxxxxx',
  'mail_smtppassword' => 'xxxxxxxxxxx',
  'theme' => '',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_smtpsecure' => 'ssl',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'has_internet_connection' => true,
  'updater.release.channel' => 'stable',
  'instanceid' => 'xxxxxx',
  'mysql.utf8mb4' => true,
  'default_language' => 'de_DE',
  'default_locale' => 'de',
  'app_install_overwrite' => 
  array (
    0 => 'admin_notifications',
    1 => 'spreed',
  ),
  'mail_sendmailmode' => 'smtp',
  'trusted_proxies' => '1.2.3.4'

);

```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

**Suggestion:**
The warning should not disappear when trusted_proxies is not set as an array

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #26256

### CODE REVIEW

File: apps/settings/tests/Controller/CheckSetupControllerTest.php [hunk 1]
```diff
@@ -337,12 +337,12 @@ public function testIsPhpSupportedTrue() {
...
 	 * @dataProvider dataForwardedForHeadersWorking
 	 *
-	 * @param array $trustedProxies
+	 * @param string|array $trustedProxies
 	 * @param string $remoteAddrNotForwarded
 	 * @param string $remoteAddr
 	 * @param bool $result
 	 */
-	public function testForwardedForHeadersWorking(array $trustedProxies, string $remoteAddrNotForwarded, string $remoteAddr, bool $result) {
+	public function testForwardedForHeadersWorking($trustedProxies, string $remoteAddrNotForwarded, string $remoteAddr, bool $result) {
```
Comment (nickvergessen) [inline apps/settings/tests/Controller/CheckSetupControllerTest.php:345]: this file is totally broken and not executed...

Comment (nickvergessen) [reply apps/settings/tests/Controller/CheckSetupControllerTest.php]: fixed

**Review by skjnldsv — APPROVED** [review]

**Review by icewind1991 — APPROVED** [review]

**Comment by nickvergessen** [comment]
Rebasing fixes psalm? lets see

**Comment by nickvergessen** [comment]
Rebasing fixes psalm? lets see

**Pull request merged**